pkg/apis/storage/types.go

```diff
@@ -337,6 +337,35 @@ type CSIDriverSpec struct {
 	//
 	// +optional
 	StorageCapacity *bool
+
+	// ServiceAccountTokens indicates the csi driver needs pods' service account
+	// tokens it is mounting volume for to do necessary authentication. Kubelet
+	// will pass the tokens in VolumeContext in the CSI NodePublishVolume calls.
+	// The CSI driver should parse and validate the following VolumeConext:
+	// "csi.storage.k8s.io/serviceAccount.tokens": {
+	//   'audience-0': {
+	//     'token': token-0,
+	//     'expiry': expiry-0,
+	//   },
+	//   // If audience is not specified, the output key will be ''
+	//   '': {
+	//     'token': token,
+	//     'expiry': expiry,
+	//   },
+	//   ...
+	// }
+	//
+	// Note: Audience in each ServiceAccountToken should be different and at
+	// most one token can ignore Audience.
+	//
+	// +optional
+	// +listType=set
+	ServiceAccountTokens []ServiceAccountToken
+
+	// RequiresRemount indicates the csi driver wants `NodePublishVolume` being
+	// periodically called to reflect any possible change in the mounted volume.
```

Comment:
How often is it called? Is it tunable?

Reply:
<= 10/s. it is the same frequency as the reconciler loop.
https://github.com/zshihang/kubernetes/blob/ebb6995f3e5fc696596502fbf861fc68c09efe4e/pkg/kubelet/volumemanager/volume_manager.go#L57-L59